### DISCUSSION TITLE
Combining GIF, part 2

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7652] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

How do I append gifs vertically and horizontally if they have the same number of frames?

**Comment by fmw42** [comment]
Unix syntax:


Horizontally:

```
magick \( image1.gif[0-N]  -coalesce +repage -append \) \
\( image2.gif[0-N] -coalesce +repage -append \) \
+append  -crop xHEIGHT +repage \
-set delay 0  animation.gif
```

Vertically:

```
magick \( image1.gif[0-N]  -coalesce +repage +append \) \
\( image2.gif[0-N] -coalesce +repage +append \) \
-append  -crop WIDTHx +repage \
-set delay 0  animation2.gif

```

where N is the number of frames you want to use (less than the minimum of the two animations, WIDTH is the width of your two animations and HEIGHT is the height of your two animations.  They must use the same number of frames and the same dimensions as used above. If they have the same number of frames, then you don't need to append the [0-N].

**Comment by snibgo** [comment]
fmw42's solution looks good.

However, that is a bash script. Windows 11 can run bash, of course. But if Windows BAT is used, change every backslash `\` to caret `^`. The parentheses don't need to be escaped with carets.

**Reply by fmw42** [reply]
Right, sorry, I forgot it was windows
